exploration_scripts/trajectory_plotting

```diff
@@ -0,0 +1,126 @@
+#!/usr/bin/env python
+from matplotlib import pyplot as plt
+import numpy as np
+import rosbag
+import sys
+
+def get_waypoints(bag_name):
+    bag = rosbag.Bag(bag_name, 'r')
+    waypoints = None
+    for topic, msg, t in bag.read_messages():
+        print "Number of waypoints: ", len(msg.goal.trajectory.joint_trajectory.points)
+        waypoints = msg.goal.trajectory.joint_trajectory.points
+    bag.close()
+
+    positions = np.array([np.array(wp.positions) for wp in waypoints])
+    velocities = np.array([np.array(wp.velocities) for wp in waypoints])
+    accelerations = np.array([np.array(wp.accelerations) for wp in waypoints])
+    times = np.array([wp.time_from_start.to_sec() for wp in waypoints])
+    return (positions, velocities, accelerations, times)
+
+def calc_derivative_from_sequence(sequence, times):
+    derivatives = np.array([0.0, 0.0, 0.0, 0.0, 0.0, 0.0])
+    for i in range(len(sequence)-1):
```

Comment:
for iterating through trajectory points, use something like `point_i`. For iterating through joints within a trajectory point, use something like `joint_i`. This makes this code clearer whey you're indexing by i.

Reply:
Done